orbit/tools/build/build-windows.go

```diff
@@ -0,0 +1,265 @@
+			PrivateBuild:     "",
+			ProductName:      "Fleet osquery",
+			ProductVersion:   version,
+			SpecialBuild:     "",
+		},
+		VarFileInfo: goversioninfo.VarFileInfo{
+			Translation: goversioninfo.Translation{
+				LangID:    goversioninfo.LangID(langID),
+				CharsetID: goversioninfo.CharsetID(charsetID),
+			},
+		},
+		IconPath:     iconPath,
+		ManifestPath: manifestPath,
+	}
+
+	return &result, nil
+}
+
+// sanitizeVersion returns the version parts (Major, Minor, Patch and Build), filling the Build part
+// with '0' if missing. Will error out if the version string is missing the Major, Minor or
+// Patch part(s).
+func sanitizeVersion(version string) ([]string, error) {
+	vParts := strings.Split(version, ".")
+	if len(vParts) < 3 {
+		return nil, errors.New("invalid version string")
+	}
+
+	if len(vParts) < 4 {
+		vParts = append(vParts, "0")
+	}
+
+	return vParts[:4], nil
+}
+
+// writeManifestXML creates the manifest.xml file used when generating the 'resource.syso' metadata
+// (see writeResourceSyso). Returns the path of the newly created file.
+func writeManifestXML(vParts []string, orbitPath string) (string, error) {
+	filePath := filepath.Join(orbitPath, "manifest.xml")
+
+	tmplOpts := struct {
+		Version string
+	}{
+		Version: strings.Join(vParts, "."),
+	}
+
+	var contents bytes.Buffer
+	if err := manifestXMLTemplate.Execute(&contents, tmplOpts); err != nil {
+		return "", fmt.Errorf("parsing manifest.xml template: %w", err)
+	}
+
+	if err := ioutil.WriteFile(filePath, contents.Bytes(), constant.DefaultFileMode); err != nil {
+		return "", fmt.Errorf("writing manifest.xml file: %w", err)
+	}
+
+	return filePath, nil
+}
+
+// Build the target binary for Windows
+func buildTargetBinary(cmdDir string, version string, binaryPath string) error {
+	var buildExec *exec.Cmd
+
+	// convert relative to full output path
+	outputBinary, err := filepath.Abs(binaryPath)
+	if err != nil {
+		return fmt.Errorf("error converting binary path to absolute: %w", err)
+	}
+
+	// check if cmdDir contains desktop
+	// if it does, add -ldflags "-H=windowsgui" to exec.Command
+	if strings.Contains(cmdDir, "desktop") {
+		linkFlags := fmt.Sprintf("-H=windowsgui -X=main.version=%s", version)
+		buildExec = exec.Command("go", "build", "-ldflags", linkFlags, "-o", outputBinary)
+	} else {
+		linkFlags := fmt.Sprintf("-X=github.com/fleetdm/fleet/v4/orbit/pkg/build.Version=%s", version)
+		buildExec = exec.Command("go", "build", "-ldflags", linkFlags, "-o", outputBinary)
+	}
+	buildExec.Env = append(os.Environ(), "GOOS=windows", "GOARCH=amd64")
+	buildExec.Stderr = os.Stderr
+	buildExec.Stdout = os.Stdout
+	buildExec.Dir = cmdDir
+
+	if err := buildExec.Run(); err != nil {
+		return fmt.Errorf("compile orbit: %w", err)
+	}
+	return nil
+}
+
+// Adapted from https://github.com/josephspurrier/goversioninfo/blob/master/testdata/resource/goversioninfo.exe.manifest
+var manifestXMLTemplate = template.Must(template.New("").Option("missingkey=error").Parse(
```

Comment:
Should we move this to `windows_template.go`?

Reply:
I agree, I just fixed this to use common template